common/test-data/src/main/java/com/csgroup/s2pdgs/data/DataConfig.java

```diff
@@ -0,0 +1,70 @@
+package com.csgroup.s2pdgs.data;
+
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileAttribute;
+import java.util.Date;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+
+import io.vavr.control.Try;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.With;
+import lombok.extern.slf4j.Slf4j;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@With
+@Slf4j
+public class DataConfig {
+    // Data location
+    private String location;
+    // data type
+    private String type;
+    
+    // i.e. DCS
+    private String prefix;
+    // 05
+    private String station;
+    // S2B
+    private String satellite;
+    
+    private String reference;
+    
+    private Function<String, String> referenceGenerator;
+    
+    private Date startDate;
+    private Date endDate;
+    // session date dir - offset => last modified time for each file and directory created
+    private int offset;
+    private int offsetUnit;
+    // number of session(s) required
+    private int session;
+    // Period between each session
+    // If both frequency and session are set, session is used instead of frequency.
+    private long frequency;
+    // Number of channel to create
+    private int channel;
+    // Number of raw file to create per channel
+    private int raw;
+    // Raw file size;
+    private long rawSize;
+
+
+    public boolean build () {
+        return Try.of(() -> Path.of(location))
+            .filter(path -> ! Files.isDirectory(path, new LinkOption[] {}))
```

Comment:
Remove the `new LinkOption[] {}` parameter

```suggestion
            .filter(path -> ! Files.isDirectory(path))
```